### PR TITLE
Add more warnings when delete key

### PR DESCRIPTION
**Pull request by kkkkun**
Fixes https://github.com/etcd-io/etcd/issues/13705

### CODE REVIEW

**Comment by kkkkun** [comment]
> Step 1/14 : FROM ubuntu:21.10
>toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
make[1]: *** [build-docker-test] Error 1
make[1]: Leaving directory `/home/runner/etcd'
make: *** [ensure-docker-test-image-exists] Error 

This failed cases doesn't seem relevant.

File: etcdctl/ctlv3/command/del_command.go [hunk 1]
```diff
@@ -46,6 +48,12 @@ func NewDelCommand() *cobra.Command {
 func delCommandFunc(cmd *cobra.Command, args []string) {
 	key, opts := getDelOp(args)
 	ctx, cancel := commandCtx(cmd)
+	getResp, err := mustClientFromCmd(cmd).Get(ctx, key, append(opts, clientv3.WithCountOnly())...)
```
Comment (ptabor) [inline etcdctl/ctlv3/command/del_command.go:51]: I think that proposal was only to make the command that uses '2 arguments' syntax safer, 
it is: 
a) introduce flag to be explicit about removing keys in a range. 
b) steer user's towards migration to the syntax with the --range. 

In 3.5 we would have only warning and a flag to opt-in.  
In 3.6 we would have some moderate sleep. Still I'm not sure we want to slow-down the operation by default by doing 'count' first. We should have flag like '--interactive=false' that disables such overheads.

Comment (kkkkun) [reply etcdctl/ctlv3/command/del_command.go]: Ok. Fixed.

Comment (kkkkun) [reply etcdctl/ctlv3/command/del_command.go]: > In 3.5 we would have only warning and a flag to opt-in. 

This PR only add warning. Should I commit PR to main or release-3.5 directly?

> In 3.6 we would have some moderate sleep. Still I'm not sure we want to slow-down the operation by default by doing 'count' first. We should have flag like '--interactive=false' that disables such overheads.

In 3.6. I  will commit another PR.

File: etcdctl/ctlv3/command/del_command.go [hunk 2]
```diff
@@ -44,6 +45,7 @@ func NewDelCommand() *cobra.Command {
 
 // delCommandFunc executes the "del" command.
 func delCommandFunc(cmd *cobra.Command, args []string) {
+	fmt.Fprintf(os.Stderr, "Warning: Data which keys match %v will be deleted.", delPrefix)
```
Comment (ptabor) [inline etcdctl/ctlv3/command/del_command.go:48]: 1. Warning should be printed only when user is doing wide-range operation, i.e. either prefix delete or range delete, not a lookup-delete. 

2.  It's good when warning is 'constructive', i.e. user can fix the 'usage' to not see the warning. but for this we need expected flag for --range deletes.

Comment (kkkkun) [reply etcdctl/ctlv3/command/del_command.go]: Thanks. You are right. 

> Warning should be printed only when user is doing wide-range operation

Fixes

**Pull request closed without merging**